libs/wingc/src/parser.rs

```diff
@@ -735,54 +801,95 @@ impl<'s> Parser<'s> {
 			None
 		};
 
-		// if the module name is a path ending in .w, create a new Parser to parse it as a new Scope,
-		// and create a StmtKind::Module instead
-		if module_name.name.starts_with("\"") && module_name.name.ends_with(".w\"") {
-			let module_path = Utf8Path::new(&module_name.name[1..module_name.name.len() - 1]);
+		let module_path = Utf8Path::new(&module_name.name[1..module_name.name.len() - 1]);
+		if is_absolute_path(&module_path) {
+			return self.with_error(
+				format!("Cannot bring \"{}\" since it is not a relative path", module_path),
+				statement_node,
+			);
+		}
+
+		if module_name.name.starts_with("\".") && module_name.name.ends_with("\"") {
 			let source_path = normalize_path(module_path, Some(&Utf8Path::new(&self.source_name)));
 			if source_path == Utf8Path::new(&self.source_name) {
 				return self.with_error("Cannot bring a module into itself", statement_node);
 			}
 			if !source_path.exists() {
-				return self.with_error(format!("Cannot find module \"{}\"", source_path), statement_node);
+				return self.with_error(format!("Cannot find module \"{}\"", module_path), statement_node);
 			}
-			if !source_path.is_file() {
-				return self.with_error(
-					format!("Cannot bring module \"{}\": not a file", source_path),
-					statement_node,
-				);
-			}
-			if source_path.ends_with(".main.w\"") || source_path.ends_with(".test.w\"") {
+
+			// case: .w file
+			if source_path.ends_with("/main.w") || source_path.ends_with(".main.w\"") || source_path.ends_with(".test.w\"") {
 				return self.with_error(
 					format!(
-						"Cannot bring module \"{}\": main/test files cannot be brought",
-						source_path
+						"Cannot bring module \"{}\": entrypoint files cannot be imported",
+						module_path
 					),
 					statement_node,
 				);
 			}
-			self.referenced_wing_files.borrow_mut().push(source_path.clone());
 
-			// parse error if no alias is provided
-			let module = if let Some(alias) = alias {
-				Ok(StmtKind::Bring {
-					source: BringSource::WingFile(Symbol {
-						name: source_path.to_string(),
-						span: module_name.span,
-					}),
-					identifier: Some(alias),
-				})
-			} else {
-				self.with_error::<StmtKind>(
-					format!(
-						"bring {} must be assigned to an identifier (e.g. bring \"foo\" as foo)",
-						module_name
-					),
-					statement_node,
-				)
-			};
+			if source_path.is_file() {
+				if source_path.extension() != Some("w") {
+					return self.with_error(
```

Comment:
silently ignore? what if there's a README.md file etc..

Reply:
What would `bring "./README.md"` do?

Reply:
oh maybe you're asking about how we handle directories generally -- we just skip trying to parse any file that doesn't have an extension of .w

https://github.com/winglang/wing/blob/bd3003689b771cc7d81646a3839573c6db43bb83/libs/wingc/src/parser.rs#L193-L203